addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4454,10 +4463,154 @@ registry.ImageOptimize = ImageHandlerOption.extend({
+            const svgImage = svg.querySelector('image');
+            img.src = svgImage.getAttribute('xlink:href');
+            img.mimetype = 'image/jpeg';
+            delete img.dataset.shape;
+            return;
+        }
+        const [module, fileName] = shape.split('/');
+        const clipShapeURL = `/${module}/static/image_shapes/${fileName}.svg`;
+        const clipShape = await (await fetch(clipShapeURL)).text();
+        const SVGParser = new DOMParser();
+        const svg = SVGParser.parseFromString(clipShape, 'image/svg+xml').documentElement;
+
+        const base64 = await applyModifications(img);
+
+        const svgImage = svg.querySelector('image');
+        svgImage.setAttribute('xlink:href', base64);
+        svg.setAttribute('width', img.naturalWidth);
+        svg.setAttribute('height', img.naturalHeight);
+        img.dataset.shape = shape;
+        await this._writeSvg(svg.outerHTML, fileName);
+        return this._applyColor(img.dataset.shapeColor, true, previewMode);
+    },
+
+    /**
+     * Applies the correct color to the shape by looking for the default palette's hexcode inside the OG svg files
+     * @param {string} color HEX Code for the color of the background shape
+     * @param {bool} newShape wether it's a new shape or not so we know to look for default colors or previous ones
+     * @param {string} previewMode state of the editor to know which colors to look for
+     */
+    async _applyColor(color, newShape, previewMode) {
+        const img = this._getImg();
+        if (!img.dataset.shape) {
+            return;
+        }
+
+        const fileName = img.dataset.shape.split('/')[1];
+        const clipShapeURL = img.src;
+        let clipShape = await (await fetch(clipShapeURL)).text();
+        let newColor = color && normalizeColor(color);
+        if (newShape) {
+            const DEFAULT_PALETTE = weUtils.DEFAULT_PALETTE;
+            const combinedColorsRegex = new RegExp(Object.values(DEFAULT_PALETTE).join('|'), 'gi');
+            if (previewMode === 'reset' && img.dataset.shapeColor) {
+                clipShape = clipShape.replace(combinedColorsRegex, img.dataset.shapeColor);
+            } else {
+                clipShape = clipShape.replace(combinedColorsRegex, match => {
+                    const colorId = Object.keys(DEFAULT_PALETTE).find(key => DEFAULT_PALETTE[key] === match.toUpperCase());
+                    return newColor = weUtils.getCSSVariableValue('o-color-' + colorId);
+                });
+            }
+        } else {
+            clipShape = clipShape.replace(img.dataset.shapeColor, newColor);
+        }
+        await this._writeSvg(clipShape, fileName);
+        if (!previewMode) {
+            img.dataset.shapeColor = newColor;
+        }
+    },
 
+    /**
+     * Replace the URL with a dataURL of the SVG converted in base64
+     * @param {string} svgText svg XML file as a string (inner content)
```

Comment:
This param description has me confused. Does it not include the svg tag? Also, svg is a subset of XML so you don't have to specify the XML part.